src/training_hub/algorithms/lora_grpo.py

```
@@ -0,0 +1,1503 @@
+    if data_path.endswith((".json", ".jsonl")):
+        return _load_local_dataset(data_path, n_train, n_val)
+
+    # HuggingFace dataset
+    logger.info("Downloading dataset '%s' (config=%s) from HuggingFace...", data_path, data_config)
+    try:
+        from datasets import load_dataset
+    except ImportError:
+        raise ImportError(
+            "The 'datasets' package is required for HuggingFace dataset loading. "
+            "Install with: pip install datasets"
+        )
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)